packages/firestore/src/core/bundle_impl.ts

```diff
@@ -207,7 +207,7 @@ export function bundleInitialProgress(
     documentsLoaded: 0,
     bytesLoaded: 0,
     totalDocuments: metadata.totalDocuments!,
-    totalBytes: metadata.totalBytes!
+    totalBytes: parseInt(metadata.totalBytes!.toString(), 10)
```

Comment:
Please use `normailizeNumber` for this: https://osscs.corp.google.com/firebase-sdk/firebase-js-sdk/+/master:packages/firestore/src/model/normalize.ts;l=73?q=normalizeNumber&ss=firebase-sdk%2Ffirebase-js-sdk

This is the function that we use in other places to deserialize large numbers.

Reply:
Cool!